### PR TITLE
Remove ALL Algolia Code

### PR DESCRIPTION
**Pull request by mstruve**
## What type of PR is this? (check all applicable)
- [x] Refactor
- [x] Optimization
- [x] Documentation Update

## Description
Remove all traces of Algolia from our codebase and add documentation for Elasticsearch. The Elasticsearch docs will likely need to be worked on as we go but I gave us what I thought was a good foundation. 

Once this is out we can remove Algolia from Heroku!

## Related Tickets & Documents
https://github.com/thepracticaldev/dev.to/projects/6#card-32955489
https://github.com/thepracticaldev/dev.to/projects/6#card-32912217

## Added tests?
- [x] no, because they aren't needed

## Added to documentation?
- [x] YES!

alt_text

### CODE REVIEW

File: .gitpod.yml [hunk 1]
```diff
@@ -18,14 +18,6 @@ tasks:
   - command: /home/gitpod/elasticsearch-7.5.2/bin/elasticsearch
   - init: cp config/sample_application.yml config/application.yml
     command: >
-      while [ -z "${ALGOLIASEARCH_APPLICATION_ID:=$(cat config/application.yml | grep ALGOLIASEARCH_APPLICATION_ID | sed 's/.*:\s*//')}" ] ||
```
Comment (rhymes) [inline .gitpod.yml:21]: yes!

File: Envfile [hunk 2]
```diff
@@ -65,13 +65,6 @@ variable :TWITTER_SECRET, :String, default: "Optional"
 variable :HONEYBADGER_API_KEY, :String, default: "Optional"
 variable :HONEYBADGER_JS_API_KEY, :String, default: "Optional"
 
-# Algolia for search (not optional)
```
Comment (rhymes) [inline Envfile:68]: bye bye bye!

File: Gemfile [hunk 3]
```diff
@@ -17,7 +17,6 @@ gem "acts_as_follower", github: "thepracticaldev/acts_as_follower", branch: "mas
 gem "addressable", "~> 2.7" # A replacement for the URI implementation that is part of Ruby's standard library
 gem "administrate", "~> 0.13" # A Rails engine that helps you put together a super-flexible admin dashboard
 gem "ahoy_email", "~> 1.1" # Email analytics for Rails
-gem "algoliasearch-rails", "~> 1.24" # Algolia Search is a hosted search engine capable of delivering real-time results from the first keystroke
```
Comment (rhymes) [inline Gemfile:20]: the end of an era :D

File: docs/backend/elasticsearch.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
+---
```
Comment (rhymes) [inline docs/backend/elasticsearch.md:1]: yay!

File: docs/internal/internal-search.md [hunk 5]
```diff
@@ -16,9 +16,9 @@ The view responsible for managing user reports can be found at
...
 
 For DEV, Ransack is being used exclusively in internal, for search problems in
-other parts of the app we use [Algolia][algolia].
+other parts of the app we use [Elasticsearch][elasticsearch].
 
 [feedback_messages]:
   https://github.com/thepracticaldev/dev.to/blob/4e41e4a2ac893fa2a6c36990cfe475858ffb086a/app/controllers/internal/feedback_messages_controller.rb#L4
 [ransack]: https://github.com/activerecord-hackery/ransack
-[algolia]: /backend/algolia
+[algolia]: /backend/elasticsearch
```
Comment (rhymes) [inline docs/internal/internal-search.md:24]: this should say `[elasticsearch]`

File: docs/technical-overview/stack.md [hunk 6]
```diff
@@ -17,8 +17,7 @@ For the Dev.to tech stack we use:
 - [_Sidekiq_](https://github.com/mperham/sidekiq) and
   [_Active Job_](https://guides.rubyonrails.org/active_job_basics.html) for
   background workers
-- [_Algolia_](https://www.algolia.com/) for search (Algolia will be phased out
-  in favor of [_Elasticsearch_](https://www.elastic.co/))
+- [_Elasticsearch_](https://www.elastic.co/) for in app searching
```
Comment (rhymes) [inline docs/technical-overview/stack.md:20]: maybe we can link https://www.elastic.co/elasticsearch/ instead of the company's homepage

**Comment by mstruve** [comment]
Good call @rhymes I didnt search for all caps ALGOLIA! Cleanups coming!

**Pull request merged**